S2VX.Game.Tests/VisualTests/EditorHoldNoteTests.cs

```diff
@@ -0,0 +1,134 @@
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Audio;
+using osu.Framework.Audio.Track;
+using osu.Framework.Graphics.Audio;
+using osu.Framework.Screens;
+using osu.Framework.Timing;
+using S2VX.Game.Editor;
+using S2VX.Game.Story;
+using S2VX.Game.Story.Note;
+using System.IO;
+
+namespace S2VX.Game.Tests.VisualTests {
+    public class EditorHoldNoteTests : S2VXTestScene {
+        [Cached]
+        private S2VXStory Story { get; set; } = new S2VXStory();
+        private EditorScreen Editor { get; set; }
+        private ScreenStack ScreenStack { get; set; } = new ScreenStack();
+
+        [Resolved]
+        private AudioManager Audio { get; set; }
+
+        private static string AudioPath { get; } = "VisualTests/EditorTests/10-seconds-of-silence.mp3";
+        private static float NoteAppearTime { get; } = 1000.0f;
+        private static float HoldDuration { get; } = 1000.0f;
+        private EditorHoldNote NoteToTest { get; set; }
+        private StopwatchClock StoryClock { get; set; }
+        private FramedClock FramedClock { get; set; }
+
+        [BackgroundDependencyLoader]
+        private void Load() {
+            var trackStream = File.OpenRead(AudioPath);
+            var track = new TrackBass(trackStream);
+            Audio.AddItem(track);
+            var drawableTrack = new DrawableTrack(track);
+            StoryClock = new StopwatchClock();
+            FramedClock = new FramedClock(StoryClock);
+            Story.Clock = FramedClock;
+            Editor = new EditorScreen(null, Story, drawableTrack);
+            ScreenStack.Push(Editor);
+            Add(ScreenStack);
+        }
+
+        // All tests will have a hold note that starts to appear in 1 second and lasts for 1 second
+        [SetUp]
+        public void Setup() {
+            Story.ApplyDefaultCommands(); // Ensure default commands are applied before using any properties
```

Comment:
Why is this needed? Story's `Update()` should automatically be called by osu!framework to do this.

Reply:
There is a race condition between `Update()` and when the test accesses the properties otherwise. If you debug the test and look at Story.Notes.FadeInTime for example, it will be 0. The whole reason why I needed to bring this out into a public function was so that I ensure the defaults are applied before I use them on the very next line.

Reply:
Refactored this so that we are using `ScheduleAfterChildren()`